linkerd/app/inbound/src/policy/store.rs

```diff
@@ -84,53 +75,57 @@ impl Store {
         S::ResponseBody:
             http::HttpBody<Data = tonic::codegen::Bytes, Error = Error> + Default + Send + 'static,
     {
-        let rxs = ports
-            .into_iter()
-            .map(|port| {
-                let discover = discover.clone();
-                let default = default.clone();
-                let rx = info_span!("watch", %port)
-                    .in_scope(|| discover.spawn_with_init(port, default.into()));
+        let (default_tx, default_rx) = watch::channel(ServerPolicy::from(default));
+        tokio::spawn(async move {
+            default_tx.closed().await;
+        });
+
+        // The initial set of documented policies never expire from the cache.
+        //
+        // Policies that are dynamically discovered at runtime will expire after
+        // `idle_timeout` to prevent holding policy watches indefinitely for
+        // ports that are generally unused.
+        let cache = {
+            let rxs = ports.into_iter().map(|port| {
+                let watch = watch.clone();
+                let default = default_rx.borrow().clone();
+                let rx =
+                    info_span!("watch", %port).in_scope(|| watch.spawn_with_init(port, default));
```

Comment:
tiny nit: `port` is a[ `u16`, which implements `tracing::Value`](https://docs.rs/tracing/latest/tracing/trait.Value.html#impl-Value-for-u16), so we don't _actually_ need to record it using `fmt::Display`:
```suggestion
                    info_span!("watch", port).in_scope(|| watch.spawn_with_init(port, default));
```